### ISSUE TITLE
checkers/importShadow_checker.go: fails to catch import shadowing in function parameters

### ISSUE THREAD

**Issue by henryleduc** [go-critic/go-critic#1494]
# `importShadow` does not detect shadowing in function parameters

## Current Behaviour
 `importShadow` catches only variable assignments, not function parameters

```go
import "context"

func main() {
    context := context.Background() // ✅ Linter triggers here
}
```

## Problem
However, it does not catch this case:
```go
import (
	"context"
	"fmt"
)

func main() {
	handler := func(context context.Context) {
		// 🚫 Linter not triggered
		fmt.Println(context)
	}
	handler(context.Background())
}
```

## Proposal
Extend the importShadow checker to catch function parameter names and trigger when the shadow an import
